README.md

```diff
@@ -1,2 +1,243 @@
+
+```bash
+cargo run --bin blockstack-cli publish b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 0 kv-store ./kv-store.clar --testnet | xxd -r -p > tx1.bin
+```
+
+### Run the testnet
+
+You can observe the state machine in action locally by running:
+
+```bash
+cargo run --bin blockstack-core testnet
+```
+
+In your console, you should observe an output with a similar:
+
+```bash
+*** mempool path: /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+The testnet is watching this directory, decoding and ingesting the transactions materialized as files. This mechanism is a shortcut for simulating a mempool. A RPC server will soon be integrated.
+
+### Publish your contract
+
+Assuming that the testnet is running, we can publish our `kv-store` contract.
+
+In another terminal (or file explorer), you can move the `tx1.bin` generated earlier, to the mempool:
+
+```bash
+cp ./tx1.bin /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+In the terminal window running the testnet, you can observe the state machine's reactions.
+
+### Reading from / Writing to the contract
+
+Now that our contract has been published on chain, let's try to submit some read / write transactions.
+We will start by trying to read the value associated with the key `foo`.
+
+To do that, we will use the subcommand:
+
+```bash
+cargo run --bin blockstack-cli contract-call --help
+```
+
+With the following arguments:
+
+```bash
+cargo run --bin blockstack-cli contract-call b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 1 ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH kv-store get-value -e \"foo\" --testnet | xxd -r -p > tx2.bin
+```
+
+And submit it by moving it to the mempool path:
+
+```bash
+cp ./tx2.bin /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+Similarly, we can generate a transaction that would be setting the key `foo` to the value `bar`:
+
+```bash
+cargo run --bin blockstack-cli contract-call b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 2 ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH kv-store set-value -e \"foo\" -e \"bar\" --testnet | xxd -r -p > tx3.bin
+```
+
+And submit it by moving it to the mempool path:
+
+```bash
+cp ./tx3.bin /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+Finally, we can issue a third transaction, reading the key `foo` again, for ensuring that the previous transaction has successfully updated the state machine:
+
+```bash
+cargo run --bin blockstack-cli contract-call b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 3 ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH kv-store get-value -e \"foo\" --testnet | xxd -r -p > tx4.bin
+```
+
+And submit this last transaction by moving it to the mempool path:
+
+```bash
+cp ./tx4.bin /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+Congratulations, you can now [write your own smart contracts with Clarity](https://docs.blockstack.org/core/smart/overview.html).
+
```

Comment:
In the tutorial, you might want to talk about the `nonce` argument a little bit so developers know to increment it on each transaction.

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1